File: algo-balanced-parentheses/python/balance_parens_spec.py
from balance_parens import balance_parens

# Add more test cases!...
print(balance_parens("abc(d)e(fgh))(i)j)k") == "abc(d)e(fgh)(i)jk")
print(balance_parens("abc((d)e(fgh)(i)j(k") == "abc(d)e(fgh)(i)jk")
print(balance_parens("") == "")
print(balance_parens(")(") == "")
print(balance_parens("(abc") == "abc")
print(balance_parens("abc)") == "abc")
print(balance_parens("()") == "()")
print(balance_parens("(((((") == "")
print(balance_parens(")())(()()(") == "()()()")
print(balance_parens(")(") == "")
print(balance_parens(")(((0)(0))((") == "((0)(0))")


# Challenge: nested parentheses...
print(balance_parens("abc(d)(ef(g(h))ij)k)lm()o)p") == "abc(d)(ef(g(h))ij)klm()op")

File: algo-balanced-parentheses/js/balanceParensSpec.js
let bp = require("./balanceParens");

// Add more test cases!...
console.log(balanceParens("abc(d)e(fgh))(i)j)k") === "abc(d)e(fgh)(i)jk")
console.log(balanceParens("abc((d)e(fgh)(i)j(k") === "abc(d)e(fgh)(i)jk")
console.log(balanceParens("") === "")
console.log(balanceParens("(abc") === "abc")
console.log(balanceParens("abc)") === "abc")
console.log(balanceParens("()") === "()")
console.log(balanceParens(")(()") === "()")
console.log(balanceParens(")(") === "")

// Challenge: nested parentheses...
console.log(balanceParens("abc(d)(ef(g(h))ij)k)lm()o)p") === "abc(d)(ef(g(h))ij)klm()op")